pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -304,20 +301,25 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		initialSync:                 atomic.NewBool(false),
 		syncTimeout:                 options.SyncTimeout,
 		discoveryNamespacesFilter:   options.DiscoveryNamespacesFilter,
+		systemNamespace:             options.SystemNamespace,
 	}
 
+	c.nsInformer = kubeClient.KubeInformer().Core().V1().Namespaces().Informer()
+	c.nsLister = kubeClient.KubeInformer().Core().V1().Namespaces().Lister()
 	if options.SystemNamespace != "" {
-		c.systemNsInformer = informers.NewSharedInformerFactoryWithOptions(c.client, options.ResyncPeriod,
-			informers.WithTweakListOptions(func(listOpts *metav1.ListOptions) {
-				listOpts.FieldSelector = fields.OneTermEqualSelector("metadata.name", options.SystemNamespace).String()
-			})).Core().V1().Namespaces().Informer()
-		c.registerHandlers(c.systemNsInformer, "Namespaces", c.onSystemNamespaceEvent, nil)
+		nsInformer := filter.NewFilteredSharedIndexInformer(func(obj interface{}) bool {
```

Comment:
It looks like this logic only cares about the system namespace, in which case we shouldn't use the `FilteredSharedIndexInformer` since that may not include the system namespace

Reply:
I donot think there is a case istio-system should be excluded. And if so, there is definitely a bug. FYI: we have envoyfilters there and there could be more global settings

Reply:
makes sense—there shouldn't be an issue because these `FilteredSharedIndexInformer` are currently only used for Services, Endpoints/EndpointSlices, and Pods. No other resources are filtered.

Reply:
DiscoverySelector has nothing to do with Envoyfilter, it impacts on service registry.

I think we need to either make systemnamespace always included in the code, or make it support istio-system filtered. Otherwise we may break folks

Reply:
I'm not quite following the thread-to clarify my suggestion, We shouldn't need to use a FilteredSharedIndexInformer for this use case, we can just use a regular namespace informer

Reply:
Yeah, the filter func works on all the namespaces.

Reply:
>Though one consideration in favor of using tweak list options may be a bit more performant since the filtering is done on the server side, whereas FilteredSharedIndexInformer does it on the client side.

For k8s apiserver, instead of establishing two list-watch streams, this can largely reduce the apiserver load in some degree.

Reply:
Is this really more efficient? It seems like a regression for 99% of users not using discoveryNamespacesFilter

Before: a single informer for a single namespace
After: a single informer for *all* namespaces, filtered server side

If we really want to do this, maybe we can only do it when discoveryNamespacesFilter is already in use?

Reply:
Unless we already use a full namespace informer for other controllers so it ends up being shared?

Reply:
discoveryNamespacesFilter is only a filter on client side based on `full namespace informer`, definitely they share same cache.

Reply:
Right but most users are not using discoveryNamespacesFilter at all I think?

But we do have the namespace controller for configmap write so I think its ok